### ISSUE TITLE
Program March 2024, Challenge 1

### ISSUE THREAD

**Issue by BeepBot99** [Mesure73L/mesure.x10.mx#92]

**Comment by BeepBot99** [comment]
@Mesure73L  This cannot be done client side. It can only be done server side. There are two options:
* Node.js
* ASP.NET
I can do the second one, but not the first one.
If you want to use Node.js, then you must do the first one.
If we use Node.js, it can be configured to be used with x10hosting.
However, x10hosting uses Apache for hosting and ASP.NET uses IIS, so you cannot host ASP.NET on x10hosting. It would have to be an API hosted on a different website, such as Microsoft Azure. On Azure, you get 10 for free, and after that, it is 20 dollars a month for unlimited. Note that it would just be an API, not a website.

**Comment by Mesure73L** [comment]
Why does this have to be server side?

**Comment by Mesure73L** [comment]
> So, what is the maximum then?
> _Originally posted by @BeepBot99 in https://github.com/Mesure73L/mesure.x10.mx/issues/60#issuecomment-1924540031_

I'm not sure, thats why I want an option in a variable.

> I'm saying, we could have a JSON file with a list of all compatible websites, and the maximum for **each indivdual website**, but that is somehow too hard for you to understand. That way, it would be more random.
> _Originally posted by @BeepBot99 in https://github.com/Mesure73L/mesure.x10.mx/issues/60#issuecomment-1924541346_

How would it be more random?

**Comment by BeepBot99** [comment]
You can not crawl websites client side for security reasons?

**Comment by Mesure73L** [comment]
> You can not crawl websites client side for security reasons?

cannot*

**Comment by BeepBot99** [comment]
> > I'm saying, we could have a JSON file with a list of all compatible websites, and the maximum for **each indivdual website**, but that is somehow too hard for you to understand. That way, it would be more random.
> > _Originally posted by @BeepBot99 in [#60 (comment)](https://github.com/Mesure73L/mesure.x10.mx/issues/60#issuecomment-1924541346)_
> 
> How would it be more random?

You would be able to have more websites in the list because there could be more than one maximum.

**Comment by Mesure73L** [comment]
> > > I'm saying, we could have a JSON file with a list of all compatible websites, and the maximum for **each indivdual website**, but that is somehow too hard for you to understand. That way, it would be more random.
> > > _Originally posted by @BeepBot99 in [#60 (comment)](https://github.com/Mesure73L/mesure.x10.mx/issues/60#issuecomment-1924541346)_
> > 
> > 
> > How would it be more random?
> 
> You would be able to have more websites in the list because there could be more than one maximum.

but it still wouldn't be more random, just more to choose from. also im gonna gather a **very big** list anyways

**Comment by Mesure73L** [comment]
Also if it can't be done client side how will the users do it?

**Comment by Mesure73L** [comment]
Also why can't you do it in node.js

**Comment by BeepBot99** [comment]
I don't know all of the libraries and stuff.

**Comment by BeepBot99** [comment]
Also, the pricing for Azure is different than it was 2 years ago, now it is 4.3 cents per API call.

(Basically nothing.)

**Comment by BeepBot99** [comment]
> Also if it can't be done client side how will the users do it?

server side
# 🤯 

Also if they don't have access to software on their computer that can do it, you can run Node.js online on some websites, such as codepen and github codespaces.

**Comment by Mesure73L** [comment]
Codepen is blocked for me and GHCS is blocked for you.

**Comment by BeepBot99** [comment]
Yes, I can run it on my computer.

**Comment by BeepBot99** [comment]
Also, I would use ASP.NET, not Node.js.

**Comment by Mesure73L** [comment]
Not everyone has a personal computer

**Comment by BeepBot99** [comment]
There are so many ways to do it online.

**Comment by BeepBot99** [comment]
I doubt Microsoft Azure is blocked for anyone. It's Microsoft.

**Comment by BeepBot99** [comment]
Also, there is [this](https://onecompiler.com/nodejs).

**Comment by Mesure73L** [comment]
So it works in any node.js code environment?

**Comment by Mesure73L** [comment]
But for the website I don't know what we should use

**Comment by Mesure73L** [comment]
If we use Azure I'd want to make there be some way to make money off of 12oC, like Google AdSense maybe.

**Comment by BeepBot99** [comment]
> So it works in any node.js code environment?

Yes

**Comment by BeepBot99** [comment]
Node.js can go directly on the website but will require a lot of configuration
ASP.NET cannot go directly on the website but will require zero to no configuration.

**Comment by BeepBot99** [comment]
Also, Azure is 4.3 cents per request.
And, Google AdSense is 1.5 dollars per 1000 views. Also, you are required to be 18 in your bank account, which you cannot lie about. So, good luck with that.

**Comment by Mesure73L** [comment]
## For the website:

I don't know, VC sometime

## For client side:

Whatever the user wants

**Comment by BeepBot99** [comment]
Can't we VC tonight?

**Comment by Mesure73L** [comment]
Yes, 18:195

**Comment by Mesure73L** [comment]
So what's the plan for this?

**Comment by BeepBot99** [comment]
I don't know, are we using ASP.NET or Node.js?

**Comment by Mesure73L** [comment]
Could we use Node.js because it would be free?

**Comment by BeepBot99** [comment]
Sure. I can do this. You can unassign yourself if you want.

**Comment by Mesure73L** [comment]
Thanks

**Comment by BeepBot99** [comment]
Done.

**Comment by Mesure73L** [comment]
It isn't done. It is not on the `develop` or `main` branches yet. Reopening this.

**Comment by Mesure73L** [comment]
I forgot, what was this challenge again?

**Comment by BeepBot99** [comment]
Calling to the API, getting a response, doing something to it, calling it again, repeating `x` times.

**Comment by Mesure73L** [comment]
Okay, what is the "doing something" part?

**Comment by BeepBot99** [comment]
Like, some operation on the input that is detmermined by the seed.

**Comment by Mesure73L** [comment]
Okay

**Comment by BeepBot99** [comment]
Yeah, so, please do this now.

**Comment by BeepBot99** [comment]
Like, seriously, stop procrastinating.

**Comment by Mesure73L** [comment]
well idrk what i need to do

**Comment by Mesure73L** [comment]
if i knew i would be doing it

**Comment by BeepBot99** [comment]
You could have said that then, please say that in the future.
How do you not know what to do?
I have told you like literally 5 times.
The first thing you need to do is to give the user their list of companies and their starting one.

**Comment by Mesure73L** [comment]
that in the future

**Comment by BeepBot99** [comment]
You know what I mean.

**Comment by BeepBot99** [comment]
> The first thing you need to do is to give the user their list of companies and their starting one.

You were so focused on the first thing I said that you didn't even realize I answered your question.
# 🤦‍♂️

**Comment by Mesure73L** [comment]
So uh, when did I say I didn't notice it??? You're just making assumptions. I did read it but didn't respond. If I responded to everything that would cause a lot of spam.

TL;DR don't assume things.

# 🤦
